tests/bdd/features/lti_certification/section_3_types_of_roles.feature

```diff
@@ -3,42 +3,115 @@ Feature: Section 3 - Types of User Roles
+
+     When I make an LTI launch request
+
+     Then the assigment opens successfully
+      And the user has instructor privileges
+
   @v1.0 @v1.1 @v1.2 @required
   Scenario: Test 3.4 - Launch as a learner using a full URN role in a list
     # Expected result: User should be given whatever privileges are appropriate
     # to a student
 
+    Given the user is 'Bob'
+      And I set the fixture 'params' key 'roles' to 'urn:non:ims/something/Else,urn:lti:role:ims/lis/Learner,urn:lti:instrole:ims/lis/Alumni'
+
+     When I make an LTI launch request
+
+     Then the assigment opens successfully
+      But the user only has learner privileges
+
   @v1.0 @v1.1 @v1.2 @required
   Scenario: Test 3.5 - Launch as a user with no context role
     # Expected result: User should be given whatever privileges are appropriate
     # to a user with no role, which could be that they are denied access
 
+    Given the user is 'Bob'
+      And I set the fixture 'params' key 'roles' to 'urn:non:ims/something/Else'
+
+     When I make an LTI launch request
+
+     Then the assigment opens successfully
+      But the user only has learner privileges
+
   @v1.0 @v1.1 @v1.2 @required
   Scenario: Test 3.6 - Launch as a user with an institution role but no context role
     # Expected result: User should be given privileges appropriate to a user
     # with no context role, which could be that they are denied access
 
+    Given I update the fixture 'params' with
+      | Key     | Value                            |
+      | roles   | urn:lti:instrole:ims/lis/Learner |
+      | user_id | user-2001                        |
+
+     When I make an LTI launch request
+
+     Then the assigment opens successfully
+      But the user only has learner privileges
+
   @v1.0 @v1.1 @v1.2 @required
   Scenario: Test 3.7 - Launch as a user with an institution role which has no corresponding context role
     # Expected result: User should be given privileges appropriate to a user
     # with no context role, which could be that they are denied access
 
+    Given I update the fixture 'params' with
+      | Key     | Value                           |
+      | roles   | urn:lti:instrole:ims/lis/Alumni |
+      | user_id | user-2001                       |
+
+     When I make an LTI launch request
+
+     Then the assigment opens successfully
+      But the user only has learner privileges
+
   @v1.0 @v1.1 @v1.2 @required
   Scenario: Test 3.8 - Launch as a user with an unrecognised role
     # Expected result: User should be given privileges appropriate to a user
-    # with no context role, which could be that they are denied access
\ No newline at end of file
+    # with no context role, which could be that they are denied access
+
+    Given I update the fixture 'params' with
+      | Key     | Value       |
+      | roles   | NotALearner |
+      | user_id | user-2001   |
+
+    When I make an LTI launch request
+
+    Then the assigment opens successfully
```

Comment:
Interesting. As things stand, we don't require any roles of our users to launch an assignment? I guess I didn't know that!